### PR TITLE
Update heightmap size & added DEM example

### PR DESCRIPTION
**Pull request by jennuine**
# 🦟 Bug fix

Requires https://github.com/ignitionrobotics/ign-common/pull/292

## Summary
Updated `heightmapSizeZ` calculation, which was not ported from gazebo-classic. Thanks to @iche033.

Also updated heightmap example to include DEMs:

```bash
cd examples
mkdir build && cd build
cmake ../
make
./heightmap dem
```

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by iche033 — COMMENTED** [review]
nice demo works for me. 

One suggestion is to keep the arguments to the heightmap example consistent with other examples, i.e.

```bash
./heightmap [engine_name] [graphics_api]
```
One idea would be to look for an `--dem` arg, e.g.

```bash
# I think your code will pick ogre 1.x by default - maybe useful to add a msg about this
./heightmap --dem

# same behavior as above
./heightmap ogre --dem

# tries to use ogre2 with dem
./heightmap ogre2 --dem

# tries to use ogre2 + metal with dem
./heightmap ogre2 metal --dem
```

File: tutorials/21_heightmap.md [hunk 1]
```diff
@@ -50,7 +60,18 @@ functionality such as:
...
+
+\snippet examples/heightmap/Main.cc create another dem heightmap
+
+Note, since DEMs encode elevation data, the minimum elevation for the `moon.tif`
+DEM is `-212.296`. We need to translate the DEM for it to be at `z = 0` by
+setting the descriptor's z position to be `212.296`
+or `desc2.SetPosition({0, 0, std::abs(data2->MinElevation())});`.
 
-\snippet examples/heightmap/Main.cc create a heightmap
+that the descriptor position
```
Comment (chapulina) [inline tutorials/21_heightmap.md:77]: This line seems out of place

**Review by chapulina — APPROVED** [review]
The demo works for me too. I like @iche033 's suggestions for the example's arguments.

**Review by chapulina — APPROVED** [review]
I've addressed mine and @iche033 's comments. I think this is good to go!

**Comment by chapulina** [comment]
@osrf-jenkins run tests please!

**Comment by chapulina** [comment]
Ahh, merging into #539 so we only iterate on CI in one place

**Pull request merged**